Feature/list dictionary data generator

aokigahara59 opened the pull request.

aokigahara59 committed:
Add TestDataGenerator; Refactor Models,Services

Added TestDataGenerators for clients, employees via list and dictionary

aokigahara59 committed:
Add name generating with Bogus

Dem-Ivan commented on BankService/TestDataGenerator.cs:
```diff
@@ -0,0 +1,66 @@
+            {
+                clients.Add(new Client
+                {
+                    Name = faker.Name.FirstName(),
```
плохо освоили работу с Богусом. Он может все!

Dem-Ivan commented on BankService/TestDataGenerator.cs:
```diff
@@ -0,0 +1,66 @@
+        {
+            var clients = new Dictionary<int, Client>();
+            Random random = new Random();
+            Bogus.Faker faker = new Bogus.Faker();
```
Вы создаете этот экземпляр в каждом методе. Разумней было бы вынести его из методов, создать один приватный.

Dem-Ivan commented on PracticeWithTypes/PracticeWithTypes.cs:
```diff
@@ -9,64 +10,76 @@ public class Program
+
+            
+            stopwatch.Start();
+            Console.WriteLine("Searching through List: "
```
1) Когда замеряйте скорость какойто операции, меряйте только время затраченное на нее. Нельзя добавлять к замеру выполнение всяких выводов в консоль.
2) Работу с sw надо вести статистически, мерять хотябы 3 раза.  Один замер не показателен.

Dem-Ivan commented on PracticeWithTypes/PracticeWithTypes.cs:
```diff
@@ -9,64 +10,76 @@ public class Program
+
+            stopwatch.Restart();
+            Console.WriteLine("Searching through Dictionary: "
+                + clientsDictionary.GetValueOrDefault(500).Name);
```
выборку по словарю делаем по ключу, иначе преимущества словаря теряются.

Dem-Ivan commented on PracticeWithTypes/PracticeWithTypes.cs:
```diff
@@ -9,64 +10,76 @@ public class Program
+            Employee lowestSalaeyEmployee = employees[0];
+            foreach (var employee in employees)
+            {
+                if (employee.Salary < lowestSalaeyEmployee.Salary) lowestSalaeyEmployee=employee;
```
что за абракадабра? Методами Linq ищем.

aokigahara59 committed:
Increase work with Bogus; StopWatch metering changes

Bogus faker for clients and employees
Using stopwach several times on one operation
Reworked searching emplyee with lowest salary
